### ISSUE TITLE
Getting invalid port for ICS-27

### ISSUE THREAD

**Issue by schnetzlerjoe** [cosmos/ibc-go#940]
## Summary of Bug

When I try to register an interchain account on two separate chains (Gaia and my Chain), the relayer throws a 

`ERROR ThreadId(115) worker.channel{channel=channel::channel-2/icacontroller-cosmos1m9l358xunhhwds0568za49mzhvuxx9uxre5tud:interamm->gaia}: Failed ChanInit with error: failed during a query for the app version to chain id gaia: failed to fetch application version: could not resolve channel version because the port is invalid: icacontroller-cosmos1m9l358xunhhwds0568za49mzhvuxx9uxre5tud. `

Cannot seem to figure it out but I do know the port name was recently changed.

## Version

v3.0.0-beta1

## Steps to Reproduce

```
git clone https://github.com/schnetzlerjoe/interamm

make install

make init

make start-rly

# Store the following account addresses within the current shell env
export DEMOWALLET_1=$(interammd keys show demowallet1 -a --keyring-backend test --home ./network/data/interamm) && echo $DEMOWALLET_1;
export DEMOWALLET_2=$(osmosisd keys show demowallet2 -a --keyring-backend test --home ./network/data/osmosis) && echo $DEMOWALLET_2;
export DEMOWALLET_3=$(gaiad keys show demowallet3 -a --keyring-backend test --home ./network/data/gaia) && echo $DEMOWALLET_3;

############ Cosmos/Gaia #################3
interammd tx intertx register --from $DEMOWALLET_1 --connection-id connection-1 --chain-id interamm --home ./network/data/interamm --node tcp://localhost:16657 --keyring-backend test -y

interammd query intertx interchainaccounts connection-1 $DEMOWALLET_1 --home ./network/data/interamm --node tcp://localhost:16657
```
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by crodriguezvega** [comment]
Thanks for opening this issue, @schnetzlerjoe.

The error message makes me wonder if the go relayer is still depending on `NegotiateAppVersion` to do version negotiation during channel handshake. We decided to remove `NegotiateAppVersion` and follow a different approach in the channel handshake. I think it would be good to check with Jack or Justin. I will share this issue on Discord.

**Comment by crodriguezvega** [comment]
I made a mistake and thought that you were using the go relayer, but now I see that you're actually using hermes...

When you had this problem, you were running version 0.9.0?

**Comment by adizere** [comment]
Hermes team recently worked on a similar issue (https://github.com/informalsystems/ibc-rs/issues/1861). The fix landed into ibc-rs@master recently.

@schnetzlerjoe if interested to use a more up-to-date version of Hermes, consider switching from 0.9.0 to building hermes from [master](https://github.com/informalsystems/ibc-rs/commits/master) branch and letting us know if that solves it. Would appreciate any feedback, thank you!

**Comment by schnetzlerjoe** [comment]
> Hermes team recently worked on a similar issue ([informalsystems/ibc-rs#1861](https://github.com/informalsystems/ibc-rs/issues/1861)). The fix landed into ibc-rs@master recently.
> 
> @schnetzlerjoe if interested to use a more up-to-date version of Hermes, consider switching from 0.9.0 to building hermes from [master](https://github.com/informalsystems/ibc-rs/commits/master) branch and letting us know if that solves it. Would appreciate any feedback, thank you!

Bingo thank you!

**Comment by schnetzlerjoe** [comment]
> Hi @schnetzlerjoe, thanks for raising this issue.
> 
> I think you must have been running hermes v0.11.0 when you encountered this, as the error itself is not actually in the hermes v0.9.0 source code.
> 
> Please see [here](https://github.com/informalsystems/ibc-rs/blob/v0.11.0/relayer/src/channel/error.rs#L77) for the error included in v0.11.0 and [here](https://github.com/informalsystems/ibc-rs/blob/v0.9.0/relayer/src/channel/error.rs#L77) for the absence of that error in v0.9.0. This will likely be removed in an upcoming release and restore the handshake behaviour of v0.9.0.
> 
> For now, I would recommend testing exclusively with v0.9.0. If you can reproduce the issue with this version we would be happy to investigate it further.

Yep! Was referencing two different cargo build locations. Thank you so much! Great catch
